core/model/modx/modx.class.php

```diff
@@ -1548,9 +1548,11 @@ public function regClientCSS($src, $media = null) {
      * tag of an HTML response.
      * @param boolean $plaintext Optional param to treat the $src as plaintext
      * rather than assuming it is JavaScript.
+     * @param string $async String literal 'async' or 'defer' which is added 
+     * as attribute to html script tag
      * @return void
      */
-    public function regClientStartupScript($src, $plaintext= false) {
+    public function regClientStartupScript($src, $plaintext= false, $async='') {
```

Comment:
Only two params allowed - `async` and `defer`, so we should add check if passed param is correct or just make it possible to turn it into a boolean parameter.

Reply:
fixed. now params checks correct values

Reply:
We should add an attributes parameter. The script tag can also have other attributes like `id`.

Reply:
There are also other methods available (e.g. `regClientHTML`) so I don't think we need an option for every possibility here.

Reply:
I was thinking about something like this:
```
public function regClientStartupScript($src, $plaintext= false, $attributes = [])
```

`$attributes = ['async', 'defer', 'id' => ['test'], 'etc'];`

Reply:
but for what? i opened w3 specs for 'script' tag:
we see:

![image](https://user-images.githubusercontent.com/8276064/55539166-b6183b00-56c8-11e9-9c15-d40fa03748c2.png)

Reply:
You are referring to W3 schools. Not the w3 specs.
[See this list of supported attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#Attributes). 
This also includes [all global attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes).